### DISCUSSION TITLE
Uncaught TypeError: carry is null

### DISCUSSION THREAD

**Discussion by mschop** [livewire/livewire#6301] [7. Bugs (archived)]
### Livewire version

3.0.0-beta8

### Which PHP version are you using?

PHP 8.2

### Repository URL

private repository

### Steps To Reproduce

1. Use eloquent model as property
2. Bind in a form element like `<text>` via `wire:model.blur` to a field of the eloquent model
3. Load the application

### Additional information

When binding to eloquent model field from an input field I get the following error message, when `'legacy_model_binding' => false` in the livewire config.

```
Uncaught TypeError: carry is null
    dataGet https://app.lndo.site/livewire/livewire.js?id=3948ee27:51
    dataGet https://app.lndo.site/livewire/livewire.js?id=3948ee27:48
    get https://app.lndo.site/livewire/livewire.js?id=3948ee27:8784
    getValue https://app.lndo.site/livewire/livewire.js?id=3948ee27:2989
    <anonymous> https://app.lndo.site/livewire/livewire.js?id=3948ee27:3045
    reactiveEffect https://app.lndo.site/livewire/livewire.js?id=3948ee27:2104
    effect2 https://app.lndo.site/livewire/livewire.js?id=3948ee27:2079
    effect https://app.lndo.site/livewire/livewire.js?id=3948ee27:490
    wrappedEffect https://app.lndo.site/livewire/livewire.js?id=3948ee27:506
    <anonymous> https://app.lndo.site/livewire/livewire.js?id=3948ee27:3044
    flushHandlers https://app.lndo.site/livewire/livewire.js?id=3948ee27:1001
    stopDeferring https://app.lndo.site/livewire/livewire.js?id=3948ee27:1006
    deferHandlingDirectives https://app.lndo.site/livewire/livewire.js?id=3948ee27:1009
    initTree https://app.lndo.site/livewire/livewire.js?id=3948ee27:1192
    start https://app.lndo.site/livewire/livewire.js?id=3948ee27:1147
    start https://app.lndo.site/livewire/livewire.js?id=3948ee27:1146
    start3 https://app.lndo.site/livewire/livewire.js?id=3948ee27:8032
    <anonymous> https://app.lndo.site/livewire/livewire.js?id=3948ee27:8949
    EventListener.handleEvent* https://app.lndo.site/livewire/livewire.js?id=3948ee27:8948
    <anonymous> https://app.lndo.site/livewire/livewire.js?id=3948ee27:8952
```

Expected behaviour:

A more specific error message should be generated. For me as a developer it was totally unclear, what the problem is.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by nam-co** [comment]
Maybe is related, but Im getting since updated to v3 release: TypeError: null is not an object (evaluating 'carry[i]')
line 347 on livewire.js
```
function dataGet(object, key) {
    if (key === "")
      return object;
    return key.split(".").reduce((carry, i) => {
      if (carry === void 0)
        return void 0;
      return carry[i];
    }, object);
  }
```

**Comment by joshhanley** [comment]
@mschop you can't bind to Eloquent Models in V3 unless you set `legacy_model_binding` to `true`. But we don't recommend using model binding, instead you should use FormObjects.

Hope this helps!

**Reply by luttje** [reply]
Do you think this problem be checked for and throw a helpful error? I spent a good hour wasting time on this. (Google hasn't indexed this thread yet)

Additionally the documentation mentions [that models and collections](https://livewire.laravel.com/docs/properties#common-php-types) are supported just fine as properties. Nowhere does it mention that databinding to them is not supported and causes this weird error.

Thanks!

**Comment by jordanade** [comment]
Even with legacy_model_binding set to true, I was still getting this issue on a page with nested and repeated components ("categories" and "items")   — but only for those _after_ the first one. 

I discovered that the fields at the end of each "category" component (with bindings to new_item.name and so forth), used to add a new "item" to the category, needed to be initialized in the mount() function. This makes sense but didn't need to be done in v2.

```
    public function mount()
    {
        $this->new_item = new Item;
    }
```
